frontend/app/src/components/section-components/chat/create-channel-form.tsx

```diff
@@ -0,0 +1,201 @@
+import { Dispatch, useState } from 'react';
+import { channelType } from '../../global-components/interface';
+import useCreateChannel from '../../query-hooks/useCreateChannel';
+
+interface CreateChannelFormProps {
+  setShowForm: Dispatch<React.SetStateAction<boolean>>;
+}
+
+const defaultFormData = {
+  name: '',
+  type: channelType.Public,
+  password: '',
+};
+
+function CreateChannelForm(props: CreateChannelFormProps) {
+  const createChannelMutation = useCreateChannel();
+  const [formData, setFormData] = useState(defaultFormData);
+  const [passwordRequired, setPasswordRequired] = useState(false);
+  const [inputStatus, setInputStatus] = useState<string>('empty');
+  const { name, password } = formData;
+
+  function onChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setInputStatus('editing');
+    setFormData((prevState) => ({
+      ...prevState,
+      [e.target.id]: e.target.value,
+    }));
+  }
+
+  function checkFormData(formData: {
+    name: string;
+    type: channelType;
+    password: string;
+  }) {
+    if (formData.name.length === 0) return 1;
+    else if (
+      formData.type === channelType.Protected &&
+      formData.password.length === 0
+    )
+      return 2;
+    return 0;
+  }
+
+  function onSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    if (checkFormData(formData) === 1) {
```

Comment:
Maybe storing error/success states in an enum could lead to more explicit code 

eg: if (checkFormData(formData) === Status.INVALIDNAME)